integr8ly/integreatly-operator
# Updating alert severity and placeholder SOPs for RHOAM GA

# Description
For RHOAM GA it has been agreed to downgrade any alert that does not directly support an SLO from `Critical` to `Warning`

Each `Critical` alert remaining requires a new SOP that meets CS-SRE minimum criteria. As a result, a series of placeholder SOPs have been created as part of https://github.com/RHCloudServices/integreatly-help/pull/710

The SOP URLs referenced in this PR are based on what the URL will look like once the changes in https://github.com/RHCloudServices/integreatly-help/pull/710 have been merged.

Applied a label of `addon: managed-api-service` to cluster monitoring alerts `RHOAMOperatorInstallDelayed` and `RHOAMUpgradeExpectedDurationExceeded` so that Service Delivery can update the Pagerduty operator to direct these alerts to CS-SRE rather than SRE-P

# Verification

NOTE: The following must be done on a CCS (BYOC) cluster

- Install RHOAM from this feature branch
- Login to prometheus
  ```
  open "https://$(oc get routes prometheus-route -n redhat-rhoam-middleware-monitoring-operator -o jsonpath='{.spec.host}')"

  ```
- For each `Critical` alert highlighted in Green on the Alerts tab of the Metrics and Monitoring [spreadsheet](https://docs.google.com/spreadsheets/d/1xqMyunLN1rccyOEKCMc6LcwMUQC-KkbH4r8zLW-95P4):
  - Ensure the alerts severity is `Critical` and has a supporting SOP in place. The URL may not be valid until https://github.com/RHCloudServices/integreatly-help/pull/710 has been merged to `master`. Also, the `RHOAMOperatorInstallDelayed` and `RHOAMUpgradeExpectedDurationExceeded` alerts are located in the cluster prometheus instance under the `openshift-monitoring` namespace
  - The `KeycloakInstanceNotAvailable` alert will not have a supporting SOP URL as we can't modify alert configurations due to this being managed in the keycloak-operator itself
- Verify that `Ratelimit-Service-Redis-RhoamRedisCacheConnectionFailed` and `Ratelimit-Service-Redis-RhoamRedisCacheUnavailable` are set to `Warning`
- Verify that all remaining alerts have a severity of either `Warning` or `Info`
- Check that the label `addon: managed-api-service` has been applied to the `RHOAMOperatorInstallDelayed` and `RHOAMUpgradeExpectedDurationExceeded` alerts:

  ```
  oc get prometheusRules rhoam-upgrade-alerts -n openshift-monitoring -o yaml
  oc get prometheusRules rhoam-installation-alerts -n openshift-monitoring -o yaml
  ```

## Checklist
